applications/ParticleMechanicsApplication/custom_strategies/schemes/MPM_residual_based_bossak_scheme.hpp

```diff
@@ -524,17 +489,11 @@ class MPMResidualBasedBossakScheme: public Scheme<TSparseSpace,TDenseSpace>
         //while (RatioNormVel > TolVel || RatioNormPres > TolPres)// && ItNum <1000)
         while (ItNum <2)
         {
-            //std::cout<< "ItNum "<<ItNum<<std::endl;
-
-//**************************************************************************************
-            
-            
             #pragma omp parallel for
 			for( int iter = 0; iter < static_cast<int>(mr_grid_model_part.Nodes().size()); ++iter)
 			{
-			
-			auto i = mr_grid_model_part.NodesBegin() + iter;
-            if( (i)->SolutionStepsDataHas(NODAL_MOMENTUM) && (i)->SolutionStepsDataHas(NODAL_MASS) && (i)->SolutionStepsDataHas(NODAL_INERTIA))//&& (i)->SolutionStepsDataHas(NODAL_INTERNAL_FORCE) )
+                auto i = mr_grid_model_part.NodesBegin() + iter;
+                if( (i)->SolutionStepsDataHas(NODAL_MOMENTUM) && (i)->SolutionStepsDataHas(NODAL_MASS) && (i)->SolutionStepsDataHas(NODAL_INERTIA))//&& (i)->SolutionStepsDataHas(NODAL_INTERNAL_FORCE) )
```

Comment:
this check is a bit strange...all nodes have the same variables, so i don't understand why it is needed

Reply:
@RiccardoRossi I have updated the codes based on your other suggestions. Though I am not sure also why this if condition is written. Maybe @iiaconeta can give some clarification?

Reply:
ok, i don't consider this a blocker. go ahead with the merging...and let's see whay @iiaconeta tells